src/AspNet.Security.OAuth.Vkontakte/VkontakteAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.Vkontakte
+{
+    /// <summary>
+    /// An ASP.NET Core middleware for authenticating users using Vkontakte
+    /// </summary>
+    public class VkontakteAuthenticationMiddleware : OAuthMiddleware<VkontakteAuthenticationOptions>
+    {
+        /// <summary>
+        /// Initializes a new <see cref="VkontakteAuthenticationMiddleware"/>.
+        /// </summary>
+        /// <param name="next">The next middleware in the HTTP pipeline to invoke.</param>
+        /// <param name="dataProtectionProvider"></param>
+        /// <param name="loggerFactory"></param>
+        /// <param name="encoder"></param>
+        /// <param name="sharedOptions"></param>
+        /// <param name="options">Configuration options for the middleware.</param>
+        public VkontakteAuthenticationMiddleware(RequestDelegate next, IDataProtectionProvider dataProtectionProvider, ILoggerFactory loggerFactory, UrlEncoder encoder, IOptions<SharedAuthenticationOptions> sharedOptions, IOptions<VkontakteAuthenticationOptions> options)
+            : base(next, dataProtectionProvider, loggerFactory, encoder, sharedOptions, options)
+        {
+            if (next == null)
```

Comment:
Remove all these null checks, they are already enforced by the base constructor.